crates/uv-resolver/src/lock/mod.rs

```diff
@@ -2617,6 +2617,14 @@ impl Package {
     fn is_dynamic(&self) -> bool {
         self.id.version.is_none()
     }
+
+    /// Returns `true` if the package contains the [`ExtraName`].
+    pub fn contains_extra(&self, name: &ExtraName) -> bool {
+        self.metadata
+            .provides_extras
+            .as_ref()
+            .is_some_and(|provides_extras| provides_extras.contains(name))
+    }
```

Comment:
I'm not sure if this is the right place to handle this but I believe this isn't our desired semantic for the case where `provides_extras` is undefined. We aren't doing a lockfile version bump, so we're still supporting `provides_extras` being missing.

Some part of this code should handle that as "I don't actually have information on if the extra is defined, and will conservatively not error, because I'm uncertain".

Reply:
Oh yeah definitely, will update the logic.